### PR TITLE
fix regression in ChannelBinding/NTLM auth

### PR DESCRIPTION
**Pull request by wfurt**
This regression caused by #32008. The old security buffer had dedicated size property. With the PR, we switched to using Span and Span.Length. However that is not correct for TokenBinding. For that the span is empty e.g. `cbBuffer` becomes 0 and OS will reject ChannelBinding buffer. 
Alternatively we can convert the UnmanagedToken to Span as well. But since currently doing not have any good test infrastructure for this, I would prefer to limit the changes.

This effectively solves #39011 but I will leave it open until we have tests to prevent regression like this. 

contributes to #39011.

### CODE REVIEW

File: src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs [hunk 1]
```diff
@@ -687,28 +709,50 @@ internal abstract partial class SafeDeleteContext : SafeHandle
...
-                            (IntPtr)pinnedToken1;
+                        if (inSecBuffers._item1.UnmanagedToken != null)
+                        {
+                            Debug.Assert(inSecBuffers._item1.Type == SecurityBufferType.SECBUFFER_CHANNEL_BINDINGS);
+                            inUnmanagedBuffer[1].pvBuffer = (IntPtr)inSecBuffers._item1.UnmanagedToken.DangerousGetHandle();
+                            inUnmanagedBuffer[1].cbBuffer = ((ChannelBinding)inSecBuffers._item1.UnmanagedToken).Size;
+                        }
+                        else
+                        {
+                            inUnmanagedBuffer[1].cbBuffer = inSecBuffers._item0.Token.Length;
```
Comment (geoffkizer) [inline src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs:737]: Shouldn't this be _item1?

And pinnedToken1 on the next line

**Comment by geoffkizer** [comment]
Aside from that, LGTM

File: src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs (hunk 1 above)
Comment (wfurt) [reply src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs]: good catch. fixed.

**Review by geoffkizer — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
@wfurt can you confirm if this change is already in .NET 5 preview 8?

**Comment by wfurt** [comment]
It does not seems to @SteveL-MSFT. Looks like the fix missed the train and I don't see it in preview8 branch.

**Comment by SteveL-MSFT** [comment]
@wfurt thanks for the confirmation.  The reported issues in the PowerShell web cmdlets repro'd with a build with .NET 5 preview 8 and was concerned that this PR didn't fix it.  Looks like I'll have to see if it's fixed with RC1.

**Comment by wfurt** [comment]
unfortunately, yes. You can always grab daily build if you have local repro outside of production.

**Pull request merged**